composer/core/state.py

```diff
@@ -505,9 +505,48 @@ def load_state_dict(self, state: Dict[str, Any], strict: bool = False):
         """
         state = _ensure_backwards_compatible_checkpointing(state)
 
+        # Verify surgery algorithms from checkpoint are also enabled in current run
+        if 'algorithms' in state:
+            import composer.algorithms as algorithms
+
+            # Get the types and counts of existing algorithms
+            existing_algorithm_counts = {}
```

Comment:
can use `defaultdict(int)` for this, e.g.
```
algo_count = defaultdict(int)
for algo in self.algorithms:
    algo_count[type(algo)] += 1
```

style -- also suggest more compact variable names.

Reply:
But also.. Do we need to count?

I don't see surgery being applied twice. We can future proof this, but actually I think the bigger edge case here is if someone applied the same surgery algorithm, but with different input parameters.

Reply:
I'd rather future proof for now